service/src/test/java/uk/nhs/adaptors/gp2gp/uat/EhrExtractUATTest.java

```diff
@@ -0,0 +1,165 @@
+    @Mock
+    private TimestampService timestampService;
+    @Mock
+    private CodeableConceptCdMapper codeableConceptCdMapper;
+
+    private EhrExtractMapper ehrExtractMapper;
+    private MessageContext messageContext;
+
+    @BeforeEach
+    public void setUp() {
+        getGpcStructuredTaskDefinition = GetGpcStructuredTaskDefinition.builder()
+            .nhsNumber(TEST_NHS_NUMBER)
+            .conversationId(TEST_CONVERSATION_ID)
+            .requestId(TEST_REQUEST_ID)
+            .fromOdsCode(TEST_FROM_ODS_CODE)
+            .toOdsCode(TEST_TO_ODS_CODE)
+            .build();
+
+        when(randomIdGeneratorService.createNewId()).thenReturn(TEST_ID_1, TEST_ID_2, TEST_ID_3);
+        when(timestampService.now()).thenReturn(Instant.parse(TEST_DATE_TIME));
+        messageContext = new MessageContext(randomIdGeneratorService);
+        EncounterComponentsMapper encounterComponentsMapper = new EncounterComponentsMapper(
+            messageContext,
+            new DiaryPlanStatementMapper(messageContext, codeableConceptCdMapper),
+            new NarrativeStatementMapper(messageContext),
+            new ObservationStatementMapper(
+                messageContext,
+                new StructuredObservationValueMapper(),
+                new PertinentInformationObservationValueMapper(),
+                codeableConceptCdMapper
+            ),
+            new ImmunizationObservationStatementMapper(messageContext, codeableConceptCdMapper),
+            new ConditionLinkSetMapper(messageContext, randomIdGeneratorService, codeableConceptCdMapper),
+            new BloodPressureMapper(
+                messageContext, randomIdGeneratorService, new StructuredObservationValueMapper(), codeableConceptCdMapper)
+        );
+
+        ehrExtractMapper = new EhrExtractMapper(randomIdGeneratorService,
+            timestampService,
+            new EncounterMapper(messageContext, encounterComponentsMapper));
+    }
+
+    @AfterEach
+    public void tearDown() {
+        messageContext.resetMessageContext();
+    }
+
+    @ParameterizedTest
+    @MethodSource("testValueFilePaths")
+    public void When_MappingProperJsonRequestBody_Expect_ProperXmlOutput(String inputJson, String expectedOutputXml) throws IOException {
+        String expectedJsonToXmlContent = ResourceTestFileUtils.getFileContent(OUTPUT_PATH + expectedOutputXml);
+        String inputJsonFileContent = ResourceTestFileUtils.getFileContent(INPUT_PATH + inputJson);
+        Bundle bundle = new FhirParseService().parseResource(inputJsonFileContent, Bundle.class);
+        messageContext.initialize(bundle);
+
+        EhrExtractTemplateParameters ehrExtractTemplateParameters = ehrExtractMapper.mapBundleToEhrFhirExtractParams(
+            getGpcStructuredTaskDefinition,
+            bundle);
+        String outputXml = ehrExtractMapper.mapEhrExtractToXml(ehrExtractTemplateParameters);
+        System.out.println(outputXml);
```

Comment:
Do we need this `system.out.println`? If not, could it be removed please?

Reply:
Oh, no, we definitely don't need it

Reply:
dfdeccbb374273ec0cf2d19752a525c9b5d564bf